tests/test_e2e.py

```diff
@@ -356,3 +358,19 @@ def test_raise_authentication_error():
 def test_raise_business_logic_error(uut):
     with pytest.raises(BusinessLogicError):
         uut.trading.orders.get_order("invalid")
+
+
+@pytest.mark.e2e
+def test_streaming_authentication(uut: Api):
+    token = uut.streaming.authenticate()
+    assert type(token) is Token
+    assert len(token.token)
+    assert len(token.user_id)
+    assert token.expires_at > datetime.now()
+
+@pytest.mark.e2e
+def test_streaming_authentication_bad_token():
+    uut = Api("bad_token", "")
```

Comment:
Do not use internals of the SDK - lets test by using its interface:

`uut = api.create('provide invalid token')`